src/robusta/core/sinks/slack/slack_sink.py

```diff
@@ -9,7 +12,141 @@ def __init__(self, sink_config: SlackSinkConfigWrapper, registry):
+            finding_data = finding.attribute_map
+            # The top level entity name (the owner of the pod etc)
+            finding_data["workload"] = finding.service.name if finding.service else None
+            status: FindingStatus = (
+                FindingStatus.RESOLVED if finding.title.startswith("[RESOLVED]") else FindingStatus.FIRING
+            )
+
+            # 1. Notification accounting
+            group_by_classification, group_by_classification_header = self.classify_finding(
+                finding_data, self.params.grouping.group_by
+            )
+            if (
+                group_by_classification in self.finding_group_start_ts
+                and timestamp - self.finding_group_start_ts[group_by_classification] > self.params.grouping.interval
+            ):
+                self.reset_grouping_data()
+            if group_by_classification not in self.finding_group_start_ts:
+                # Create a new group/thread
+                self.finding_group_start_ts[group_by_classification] = timestamp
+                slack_thread_ts = None
+            else:
+                slack_thread_ts = self.finding_group_heads.get(group_by_classification)
+            self.finding_group_n_received[group_by_classification] += 1
+            if (
+                not self.grouping_summary_mode
+                and self.finding_group_n_received[group_by_classification]
+                < self.params.grouping.notification_mode.regular.ignore_first
+            ):
+                return
+
+            if self.grouping_summary_mode:
+                summary_classification, summary_classification_header = self.classify_finding(
+                    finding_data, self.params.grouping.notification_mode.summary.by
+                )
+
+            # 2. Notification sending
+            if slack_thread_ts is not None:
+                # Continue emitting findings in an already existing Slack thread
+                if self.grouping_summary_mode:
+                    idx = 0 if status == FindingStatus.FIRING else 0
+                    self.finding_summary_counts[group_by_classification][summary_classification][idx] += 1
+                    # Update the summary message
+                    self.slack_sender.send_or_update_summary_message(
+                        group_by_classification_header,
+                        self.finding_summary_header,
+                        self.finding_summary_counts[group_by_classification],
+                        self.params,
+                        platform_enabled,
+                        summary_start=self.finding_group_start_ts[group_by_classification],
+                        threaded=self.params.grouping.notification_mode.summary.threaded,
+                        msg_ts=slack_thread_ts,
+                        investigate_uri=investigate_uri,
+                    )
+                if not self.grouping_summary_mode or self.params.grouping.notification_mode.summary.threaded:
```

Comment:
not sure I understand this `if`

Reply:
In case we are not using the summary mode at all (which means we use the regular mode), or in case we are using summary mode with "threaded" enabled, we we send the finding as a message to slack, into the Slack thread identified by `slack_thread_ts`

Reply:
Note: this is outdated, this part of code has been completely rewritten.